candlepin/subscription-manager
# Functional tests of yum/dnf plugins

* Just few Ansible playbooks for testing yum/dnf plugins.
* How to test it?
  - Edit /etc/ansible/hosts and add at least one machine to the
    group called clients. E.g.:

     [clients]
     centos7.example.com

  - Note: you have to be able to login to machines in group
    using ssh key (not password).
  - Execute:

```
     $ cd test/functional_tests/ansible_playbooks
     $ ansible-playbook ./configure_package_manager.yml \
         --extra-vars="candlepin_hostname=<IP_OF_TESTING_CANDLEPIN>"
     $ ansible-playbook ./register_system.yml
     $ ansible-playbook ./attach_subscriptions.yml
     $ ansible-playbook ./test_install_remove_packages.yml
     $ ansible-playbook ./test_not_remove_prod_cert_for_disabled_repo.yml
```
  - Note: ansible playbooks test_*.yml should pass after #2085 when
    is merged
